File: api/internal/app/interfaces/repo/otomo.go
package repo

import (
	"context"
	"otomo/internal/app/model"
)

type OtomoRepository interface {
	GetByUserID(ctx context.Context, userID model.UserID) (*model.Otomo, error)
	Save(ctx context.Context, otomo *model.Otomo) error
}

File: frontend/lib/repositories/otomo.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:injectable/injectable.dart';
import 'package:otomo/domains/entities/otomo.dart';
import 'package:otomo/domains/repo/otomo.dart';

@Injectable(as: OtomoRepository)
class OtomoRepositoryImpl implements OtomoRepository {
  const OtomoRepositoryImpl(this._firestore);

  final FirebaseFirestore _firestore;

  @override
  Future<Otomo> get(String userId) {
    // TODO: implement get
    throw UnimplementedError();
  }

  @override
  Future<void> save(Otomo otomo) {
    // TODO: implement save
    throw UnimplementedError();
  }
}
